components/MetadataCreation/AudioPlayer.tsx

```diff
@@ -1,18 +1,46 @@
 import { useMomentCreateProvider } from "@/providers/MomentCreateProvider/MomentCreateProvider";
-import { useRef, useState } from "react";
+import { useRef, useState, useEffect, useMemo } from "react";
 import { Button } from "@/components/ui/button";
 import { Pause, Play } from "lucide-react";
 import { Slider } from "@/components/ui/slider";
-import { getFetchableUrl } from "@/lib/protocolSdk/ipfs/gateway";
 import Image from "next/image";
 import { useMomentFormProvider } from "@/providers/MomentFormProvider";
 
 const AudioPlayer = ({ onClick }: { onClick: () => void }) => {
   const { createdContract } = useMomentCreateProvider();
-  const { imageUri, animationUri } = useMomentFormProvider();
+  const { animationFile, previewFile } = useMomentFormProvider();
   const [isPlaying, setIsPlaying] = useState(false);
   const [progress, setProgress] = useState(0);
   const audioRef = useRef<HTMLAudioElement>(null);
+  const [previewFileUrl, setPreviewFileUrl] = useState<string>("");
+
+  // Create blob URL from previewFile for cover image
+  useEffect(() => {
+    if (previewFile) {
+      const blobUrl = URL.createObjectURL(previewFile);
+      setPreviewFileUrl(blobUrl);
+      return () => URL.revokeObjectURL(blobUrl);
+    } else {
+      setPreviewFileUrl("");
+    }
+  }, [previewFile]);
+
+  // Create blob URL from animationFile
+  const audioSrc = useMemo(() => {
+    if (animationFile) {
+      return URL.createObjectURL(animationFile);
+    }
+    return "";
+  }, [animationFile]);
+
+  // Clean up blob URL on unmount
+  useEffect(() => {
+    return () => {
+      if (audioSrc && audioSrc.startsWith("blob:")) {
+        URL.revokeObjectURL(audioSrc);
+      }
+    };
+  }, [audioSrc]);
```

Comment:
Open Closed Principle
- actual: adding new state and hooks to an existing react component
- required: make a new hook for this new state / hook and import it into the AudioPlayer react component.